packages/core/admin/admin/src/pages/SettingsPage/pages/Roles/ListPage/index.js

```diff
@@ -244,17 +183,22 @@ const useRoleActions = ({ getData, canCreate, canDelete, canUpdate }) => {
 
 const RoleListPage = () => {
   const { formatMessage } = useIntl();
+  useFocusWhenNavigate();
+  const permissions = useSelector(selectAdminPermissions);
 
   const {
-    isLoadingForPermissions,
-    canCreate,
-    canRead,
-    canDelete,
-    canUpdate,
+    isLoading: isLoadingForPermissions,
+    allowedActions: { canCreate, canDelete, canRead, canUpdate },
+  } = useRBAC(permissions.settings.roles);
+
+  const {
+    roles,
     isLoading,
-    getData,
-    sortedRoles,
-  } = useSortedRoles();
+    refetch: refetchRoles,
+  } = useAdminRoles(undefined, {
```

Comment:
@jhoward1994 I've had a quick chat with Marc and he thinks it would be quite easy: could you ad filtering capabilities on this endpoint for me, please? :)

Endpoint: `/admin/settings/roles`

Reply:
Jamie's PR: https://github.com/strapi/strapi/pull/17548